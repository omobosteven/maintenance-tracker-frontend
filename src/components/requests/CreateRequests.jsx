import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import TextField from '../common/TextField';
import ValidateRequestInput from '../../validations/ValidateRequestInput';
import ErrorAlertNotification from '../common/ErrorAlertNotification';
import createRequestAction, {
  deleteErrorMessages
} from '../../actions/createRequest.action';

export class createRequestForm extends Component {
  state = {
    type: '1',
    category: 'computers',
    description: '',
    item: '',
    errors: {}
  }

  onChange = (event) => {
    const { errors } = this.state;
    if (errors[event.target.name]) {
      const newErrors = Object.assign({}, errors);
      delete newErrors[event.target.name];
      this.setState({
        [event.target.name]: event.target.value,
        errors: newErrors
      });
    } else {
      this.setState({
        [event.target.name]: event.target.value
      });
    }
  };

  onSubmit = (event) => {
    const { action, history } = this.props;
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      action.createRequest(this.state, history);
    }
  };

  handleDelete = () => {
    const { action } = this.props;
    action.deleteError();
  };

  isValid = () => {
    const { errors, isValid } = ValidateRequestInput.createInput(
      this.state
    );

    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  };

  render() {
    const { error } = this.props;
    const {
      type, category, item, description, errors
    } = this.state;

    const types = [{ id: 1, type: 'Repair' }, { id: 2, type: 'Maintenance' }];
    const categories = [
      { id: 'computers', category: 'Computers' },
      { id: 'electronics', category: 'Electronics' },
      { id: 'furnitures', category: 'Furniture' }
    ];

    return (
      <section className="section-form container">
        {
          (error && error.message) ? (
            <ErrorAlertNotification
              errors={error.message}
              onClick={this.handleDelete}
            />
          ) : ''
        }
        <div className="row card-form">
          <form className="signup-form">
            <h2 className="form-title">Create a Request</h2>
            <div className="form-group">
              <label htmlFor="type">Type</label>
              <select
                className="form-control"
                name="type"
                onChange={this.onChange}
                value={type}
              >
                {
                  types.map(data => (
                    <option key={data.id} value={data.id}>
                      {data.type}
                    </option>))
                }
              </select>
            </div>
            <p className="error">{errors.type}</p>
            <div className="form-group">
              <label htmlFor="category">Category</label>
              <select
                className="form-control"
                name="category"
                onChange={this.onChange}
                value={category}
              >
                {
                  categories.map(data => (
                    <option key={data.id} value={data.id}>
                      {data.category}
                    </option>))
                }
              </select>
            </div>
            <p className="error">{errors.type}</p>
            <TextField
              error={errors.item}
              label="Item"
              field="item"
              value={item}
              onChange={this.onChange}
              placeholder="Enter item"
              type="text"
            />
            <div className="form-group">
              <label htmlFor="description">Description</label>
              <textarea
                className="form-control"
                name="description"
                value={description}
                onChange={this.onChange}
                placeholder="Enter a description of the problem"
                rows="4"
                data-gramm_editor="false"
              />
            </div>
            { errors.description && <p className="display-error">{errors.description}</p> }
            <div className="submit-btn create-btn">
              <input type="submit" value="Create Request" onClick={this.onSubmit} />
            </div>
          </form>
        </div>
      </section>
    );
  }
}

createRequestForm.propTypes = {
  history: PropTypes.instanceOf(Object),
  error: PropTypes.instanceOf(Object),
  action: PropTypes.instanceOf(Object)
};

const mapStateToProps = state => ({
  error: state.requests.error
});

const matchDispatchToProps = dispatch => ({
  action: bindActionCreators(
    {
      createRequest: createRequestAction,
      deleteError: deleteErrorMessages
    },
    dispatch
  )
});

export default connect(mapStateToProps, matchDispatchToProps)(createRequestForm);
