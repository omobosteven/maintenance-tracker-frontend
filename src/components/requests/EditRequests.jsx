import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import TextField from '../common/TextField';
import ValidateRequestInput from '../../validations/ValidateRequestInput';
import ErrorAlertNotification from '../common/ErrorAlertNotification';
import updateRequestAction, {
  deleteErrorMessages
} from '../../actions/editRequest.action';
import fetchSingleRequestAction from '../../actions/fetchSingleRequest.action';

export class EditRequestForm extends Component {
  state = {
    type: '',
    category: '',
    description: '',
    item: '',
    errors: {}
  }

  onChange = (event) => {
    const { errors } = this.state;
    if (errors[event.target.name]) {
      const newErrors = Object.assign({}, errors);
      delete newErrors[event.target.name];
      this.setState({
        [event.target.name]: event.target.value,
        errors: newErrors
      });
    } else {
      this.setState({
        [event.target.name]: event.target.value
      });
    }
  };

  onSubmit = (event) => {
    const { action, history, match } = this.props;
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      action.updateRequest(this.state, match.params.id, history);
    }
  };

  onCancel = () => {
    const { history, match } = this.props;
    history.push(`/users/requests/${match.params.id}`);
  };

  componentDidMount = () => {
    const { match, action } = this.props;
    action.fetchRequest(match.params.id).then((res) => {
      const {
        typeId, category, item, description
      } = res.data.data.request;
      this.setState({
        type: `${typeId}`,
        category,
        item,
        description
      });
    });
  }

  handleDelete = () => {
    const { action } = this.props;
    action.deleteError();
  };

  isValid = () => {
    const { errors, isValid } = ValidateRequestInput.createInput(
      this.state
    );

    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  };

  render() {
    const { error } = this.props;
    const {
      type, category, item, description, errors
    } = this.state;

    const types = [{ id: 1, type: 'Repair' }, { id: 2, type: 'Maintenance' }];
    const categories = [
      { id: 'computers', category: 'Computers' },
      { id: 'electronics', category: 'Electronics' },
      { id: 'furnitures', category: 'Furniture' }
    ];

    return (
      <section className="section-form container">
        {
          (error && error.message) ? (
            <ErrorAlertNotification
              errors={error.message}
              onClick={this.handleDelete}
            />
          ) : ''
        }
        <div className="row card-form">
          <form className="signup-form">
            <h2 className="form-title">Modify Request</h2>
            <div className="form-group">
              <label htmlFor="type">Type</label>
              <select
                className="form-control"
                name="type"
                onChange={this.onChange}
                value={type}
              >
                {
                  types.map(data => (
                    <option key={data.id} value={data.id}>
                      {data.type}
                    </option>))
                }
              </select>
            </div>
            <p className="error">{errors.type}</p>
            <div className="form-group">
              <label htmlFor="category">Category</label>
              <select
                className="form-control"
                name="category"
                onChange={this.onChange}
                value={category}
              >
                {
                  categories.map(data => (
                    <option key={data.id} value={data.id}>
                      {data.category}
                    </option>))
                }
              </select>
            </div>
            <p className="error">{errors.type}</p>
            <TextField
              error={errors.item}
              label="Item"
              field="item"
              value={item}
              onChange={this.onChange}
              placeholder="Enter item"
              type="text"
            />
            <div className="form-group">
              <label htmlFor="description">Description</label>
              <textarea
                className="form-control"
                name="description"
                value={description}
                onChange={this.onChange}
                placeholder="Enter a description of the problem"
                rows="4"
                data-gramm_editor="false"
              />
            </div>
            { errors.description && <p className="display-error">{errors.description}</p> }
            <div className="submit-btn update-btn">
              <input type="submit" value="Modify" onClick={this.onSubmit} />
              <input type="button" value="Cancel" onClick={this.onCancel} />
            </div>
          </form>
        </div>
      </section>
    );
  }
}

EditRequestForm.propTypes = {
  history: PropTypes.instanceOf(Object),
  match: PropTypes.instanceOf(Object),
  error: PropTypes.instanceOf(Object),
  action: PropTypes.instanceOf(Object)
};

const mapStateToProps = state => ({
  error: state.requests.error
});

const matchDispatchToProps = dispatch => ({
  action: bindActionCreators(
    {
      updateRequest: updateRequestAction,
      fetchRequest: fetchSingleRequestAction,
      deleteError: deleteErrorMessages
    },
    dispatch
  )
});

export default connect(mapStateToProps, matchDispatchToProps)(EditRequestForm);
