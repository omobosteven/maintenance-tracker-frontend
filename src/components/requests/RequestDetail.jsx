import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import cap from 'capitalize';
import PropTypes from 'prop-types';
import Loading from '../common/Loading';
import fetchSingleRequestAction from '../../actions/fetchSingleRequest.action';
import routes from '../../constants/routes';

class RequestDetails extends Component {
  componentDidMount = () => {
    const { fetchRequest, match } = this.props;
    fetchRequest(match.params.id);
  }

  render() {
    const { request } = this.props;

    if (!request) {
      return (
        <div>
          <div className="row">
            <Loading />
          </div>
        </div>
      );
    }

    const getStatusText = {
      statusType: '',
      statusClassname: ''
    };

    switch (request.statusId) {
      case 1:
        getStatusText.statusType = 'Pending';
        getStatusText.statusClassname = 'request-pending';
        break;
      case 2:
        getStatusText.statusType = 'Approved';
        getStatusText.statusClassname = 'request-approved';
        break;
      case 3:
        getStatusText.statusType = 'Disapproved';
        getStatusText.statusClassname = 'request-disapproved';
        break;
      case 4:
        getStatusText.statusType = 'Resolved';
        getStatusText.statusClassname = 'request-resolved';
        break;
      default:
    }

    return (
      <section className="section-details container">
        <div className="row">
          <h2 className="section-heading">Request Details</h2>
        </div>
        <div className="row card-details">
          <Link to={routes.USER_REQUESTS} className="close-link">X</Link>
          { request.statusId === 1
              && (
                <Link to={`/edit-request/${request.requestId}`} className="edit-link">
                  <i className="mdi mdi-pencil-box-outline" />
                  Modify
                </Link>)
          }
          <div className="details user-details">
            <p>
              <span className="details-heading">Ref_no:</span>
              <span className="detail">{`#${request.ref_no}`}</span>
            </p>
            <p>
              <span className="details-heading">Type:</span>
              <span className="detail">{request.type === 1 ? 'Repair' : 'Maintenance'}</span>
            </p>
            <p>
              <span className="details-heading">Category:</span>
              <span className="detail">{cap(request.category)}</span>
            </p>
            <p>
              <span className="details-heading">Item:</span>
              <span className="detail">{cap(request.item)}</span>
            </p>
            <p>
              <span className="details-heading">Description:</span>
              <span className="detail">{request.description}</span>
            </p>
          </div>
          <div className="request-status">
            <p className={getStatusText.statusClassname}>{getStatusText.statusType}</p>
          </div>
        </div>
      </section>
    );
  }
}

RequestDetails.propTypes = {
  fetchRequest: PropTypes.func.isRequired,
  request: PropTypes.instanceOf(Object),
  match: PropTypes.instanceOf(Object)
};

const mapStateToProps = state => ({
  request: state.requests.request,
});

const matchDispatchToProps = dispatch => bindActionCreators({
  fetchRequest: fetchSingleRequestAction,
}, dispatch);

export default connect(mapStateToProps, matchDispatchToProps)(RequestDetails);
