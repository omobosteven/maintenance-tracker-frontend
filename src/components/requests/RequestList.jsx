import React from 'react';
import { Link } from 'react-router-dom';
import cap from 'capitalize';
import PropTypes from 'prop-types';

const RequestList = ({
  id, refNo, item, status, type
}) => {
  const requestClassName = {
    statusType: '',
    statusClassname: ''
  };

  switch (status) {
    case 1:
      requestClassName.statusType = 'pending';
      requestClassName.statusClassname = 'status-pending';
      break;
    case 2:
      requestClassName.statusType = 'approved';
      requestClassName.statusClassname = 'status-approved';
      break;
    case 3:
      requestClassName.statusType = 'disapproved';
      requestClassName.statusClassname = 'status-disapproved';
      break;
    case 4:
      requestClassName.statusType = 'resolved';
      requestClassName.statusClassname = 'status-resolved';
      break;
    default:
  }

  return (
    <li className={`request ${requestClassName.statusType}`}>
      <Link to={`/requests/${id}`} className="request-item">
        <p>{`#${refNo}`}</p>
        <p>{type === 1 ? 'Repair' : 'Maintenance'}</p>
        <p>{cap(item)}</p>
        <p className={requestClassName.statusClassname}>{cap(requestClassName.statusType)}</p>
      </Link>
    </li>
  );
};

RequestList.propTypes = {
  id: PropTypes.number,
  refNo: PropTypes.number,
  item: PropTypes.string,
  status: PropTypes.number,
  type: PropTypes.number
};

export default RequestList;
