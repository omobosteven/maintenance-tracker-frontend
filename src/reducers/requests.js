import {
  CREATE_REQUEST,
  CREATE_REQUEST_FAIL,
  GET_REQUESTS,
  GET_REQUESTS_FAIL,
  DELETE_REQUEST_ERROR_MESSAGE
} from '../constants/types';

const initialState = {
  requests: [],
  request: null,
  error: {}
};

export default (state = initialState, action) => {
  switch (action.type) {
    case CREATE_REQUEST:
      return {
        ...state,
        created: action.request
      };

    case CREATE_REQUEST_FAIL:
      return {
        ...state,
        error: action.error
      };

    case GET_REQUESTS:
      return {
        ...state,
        requests: action.requests
      };

    case GET_REQUESTS_FAIL:
      return {
        ...state,
        requests: []
      };

    case DELETE_REQUEST_ERROR_MESSAGE:
      return {
        ...state,
        error: {}
      };

    default: return state;
  }
};
