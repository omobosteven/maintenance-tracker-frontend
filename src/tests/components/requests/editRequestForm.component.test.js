import React from 'react';
import { shallow } from 'enzyme';
import { EditRequestForm } from '../../../components/requests/EditRequests';
import mockData from '../../__mocks__/mockData';

let wrapper;

const setProps = {
  error: {},
  action: {
    updateRequest: jest.fn(),
    fetchRequest: jest.fn().mockResolvedValue(Promise.resolve()),
    deleteError: jest.fn
  },
  match: {
    params: { id: 1 }
  },
  history: {
    push: jest.fn
  }
};

describe('EditRequestForm component', () => {
  beforeEach(() => {
    const props = { ...setProps };
    wrapper = shallow(<EditRequestForm {...props} />);
  });

  it('should render the EditRequestForm component', () => {
    const props = { ...setProps };

    wrapper = shallow(<EditRequestForm {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists()).toBe(true);
  });

  it('should set item when item changes', () => {
    const event = {
      target: {
        name: 'item',
        value: 'laptop'
      }
    };

    const item = wrapper.find('.item');
    item.simulate('change', event);

    const expectedItem = 'laptop';

    expect(wrapper.instance().state.item).toBe(expectedItem);
  });

  it('should set description when descriprion changes after receiving error', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'description',
        value: 'laptop'
      }
    };

    const description = wrapper.find('.item');

    description.simulate('change', event);

    wrapper.setState({ errors: { [event.target.name]: ['mock'] } });
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state[event.target.name])
      .toBe(event.target.value);
  });

  it('should update request when correct creadentials are supplied', () => {
    const { createRequestData } = mockData;
    const event = {
      preventDefault: jest.fn(),
    };

    const updateRequest = wrapper.find('.submit');

    wrapper.setState(createRequestData);
    updateRequest.simulate('click', event);

    expect(wrapper.instance().state.item).toBe(createRequestData.item);
  });

  it('should return error for invalid form submission', () => {
    wrapper.find('.submit').simulate('click', {
      preventDefault: jest.fn(),
    });
    expect(wrapper.state('error')).not.toBe({});
    expect(wrapper).toMatchSnapshot();
  });

  it('should delete error message if there is an error', () => {
    const props = {
      ...setProps,
      error: {
        message: 'There is an error'
      }
    };
    wrapper = shallow(<EditRequestForm {...props} />);

    const deleteErrorMessage = wrapper.find('#errorNotify');
    deleteErrorMessage.simulate('click');

    expect(wrapper).toMatchSnapshot();
  });


  it('should cancel update', () => {
    const props = {
      ...setProps
    };
    wrapper = shallow(<EditRequestForm {...props} />);

    const cancelUpdate = wrapper.find('.cancel');
    cancelUpdate.simulate('click');

    expect(wrapper).toMatchSnapshot();
  });
});
