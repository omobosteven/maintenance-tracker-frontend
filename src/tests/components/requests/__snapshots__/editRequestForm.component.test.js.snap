// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRequestForm component should cancel update 1`] = `
<section
  className="section-form container"
>
  <div
    className="row card-form"
  >
    <form
      className="signup-form"
    >
      <h2
        className="form-title"
      >
        Modify Request
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="type"
        >
          Type
        </label>
        <select
          className="form-control"
          name="type"
          onChange={[Function]}
          value=""
        >
          <option
            key="1"
            value={1}
          >
            Repair
          </option>
          <option
            key="2"
            value={2}
          >
            Maintenance
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control"
          name="category"
          onChange={[Function]}
          value=""
        >
          <option
            key="computers"
            value="computers"
          >
            Computers
          </option>
          <option
            key="electronics"
            value="electronics"
          >
            Electronics
          </option>
          <option
            key="furnitures"
            value="furnitures"
          >
            Furniture
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <TextField
        className="item"
        field="item"
        label="Item"
        onChange={[Function]}
        placeholder="Enter item"
        type="text"
        value=""
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control description"
          data-gramm_editor="false"
          name="description"
          onChange={[Function]}
          placeholder="Enter a description of the problem"
          rows="4"
          value=""
        />
      </div>
      <div
        className="submit-btn update-btn"
      >
        <input
          className="submit"
          onClick={[Function]}
          type="submit"
          value="Modify"
        />
        <input
          className="cancel"
          onClick={[Function]}
          type="button"
          value="Cancel"
        />
      </div>
    </form>
  </div>
</section>
`;

exports[`EditRequestForm component should delete error message if there is an error 1`] = `
<section
  className="section-form container"
>
  <ErrorAlertNotification
    errors="There is an error"
    id="errorNotify"
    onClick={[Function]}
  />
  <div
    className="row card-form"
  >
    <form
      className="signup-form"
    >
      <h2
        className="form-title"
      >
        Modify Request
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="type"
        >
          Type
        </label>
        <select
          className="form-control"
          name="type"
          onChange={[Function]}
          value=""
        >
          <option
            key="1"
            value={1}
          >
            Repair
          </option>
          <option
            key="2"
            value={2}
          >
            Maintenance
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control"
          name="category"
          onChange={[Function]}
          value=""
        >
          <option
            key="computers"
            value="computers"
          >
            Computers
          </option>
          <option
            key="electronics"
            value="electronics"
          >
            Electronics
          </option>
          <option
            key="furnitures"
            value="furnitures"
          >
            Furniture
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <TextField
        className="item"
        field="item"
        label="Item"
        onChange={[Function]}
        placeholder="Enter item"
        type="text"
        value=""
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control description"
          data-gramm_editor="false"
          name="description"
          onChange={[Function]}
          placeholder="Enter a description of the problem"
          rows="4"
          value=""
        />
      </div>
      <div
        className="submit-btn update-btn"
      >
        <input
          className="submit"
          onClick={[Function]}
          type="submit"
          value="Modify"
        />
        <input
          className="cancel"
          onClick={[Function]}
          type="button"
          value="Cancel"
        />
      </div>
    </form>
  </div>
</section>
`;

exports[`EditRequestForm component should render the EditRequestForm component 1`] = `
<section
  className="section-form container"
>
  <div
    className="row card-form"
  >
    <form
      className="signup-form"
    >
      <h2
        className="form-title"
      >
        Modify Request
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="type"
        >
          Type
        </label>
        <select
          className="form-control"
          name="type"
          onChange={[Function]}
          value=""
        >
          <option
            key="1"
            value={1}
          >
            Repair
          </option>
          <option
            key="2"
            value={2}
          >
            Maintenance
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control"
          name="category"
          onChange={[Function]}
          value=""
        >
          <option
            key="computers"
            value="computers"
          >
            Computers
          </option>
          <option
            key="electronics"
            value="electronics"
          >
            Electronics
          </option>
          <option
            key="furnitures"
            value="furnitures"
          >
            Furniture
          </option>
        </select>
      </div>
      <p
        className="error"
      />
      <TextField
        className="item"
        field="item"
        label="Item"
        onChange={[Function]}
        placeholder="Enter item"
        type="text"
        value=""
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control description"
          data-gramm_editor="false"
          name="description"
          onChange={[Function]}
          placeholder="Enter a description of the problem"
          rows="4"
          value=""
        />
      </div>
      <div
        className="submit-btn update-btn"
      >
        <input
          className="submit"
          onClick={[Function]}
          type="submit"
          value="Modify"
        />
        <input
          className="cancel"
          onClick={[Function]}
          type="button"
          value="Cancel"
        />
      </div>
    </form>
  </div>
</section>
`;

exports[`EditRequestForm component should return error for invalid form submission 1`] = `
<section
  className="section-form container"
>
  <div
    className="row card-form"
  >
    <form
      className="signup-form"
    >
      <h2
        className="form-title"
      >
        Modify Request
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="type"
        >
          Type
        </label>
        <select
          className="form-control"
          name="type"
          onChange={[Function]}
          value=""
        >
          <option
            key="1"
            value={1}
          >
            Repair
          </option>
          <option
            key="2"
            value={2}
          >
            Maintenance
          </option>
        </select>
      </div>
      <p
        className="error"
      >
        The type field is required.
      </p>
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control"
          name="category"
          onChange={[Function]}
          value=""
        >
          <option
            key="computers"
            value="computers"
          >
            Computers
          </option>
          <option
            key="electronics"
            value="electronics"
          >
            Electronics
          </option>
          <option
            key="furnitures"
            value="furnitures"
          >
            Furniture
          </option>
        </select>
      </div>
      <p
        className="error"
      >
        The type field is required.
      </p>
      <TextField
        className="item"
        error={
          Array [
            "The item field is required.",
          ]
        }
        field="item"
        label="Item"
        onChange={[Function]}
        placeholder="Enter item"
        type="text"
        value=""
      />
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control description"
          data-gramm_editor="false"
          name="description"
          onChange={[Function]}
          placeholder="Enter a description of the problem"
          rows="4"
          value=""
        />
      </div>
      <p
        className="display-error"
      >
        The description field is required.
      </p>
      <div
        className="submit-btn update-btn"
      >
        <input
          className="submit"
          onClick={[Function]}
          type="submit"
          value="Modify"
        />
        <input
          className="cancel"
          onClick={[Function]}
          type="button"
          value="Cancel"
        />
      </div>
    </form>
  </div>
</section>
`;
